scrypto/src/math/integer/bits.rs

```diff
@@ -0,0 +1,768 @@
+                    #[doc = "assert_eq!(m, I16::tfrom(21760i16));"]
+                    /// ```
+                    #[inline]
+                    #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+                    pub fn swap_bytes(self) -> Self {
+                        $t(self.0.into_iter().rev().collect::<Vec<u8>>().try_into().unwrap())
+                    }
+
+                    /// Reverses the bit pattern of the integer.
+                    ///
+                    /// # Examples
+                    ///
+                    /// Please note that this example is shared between large integer types.
+                    /// Which explains why `I16` is used here.
+                    ///
+                    /// Basic usage:
+                    ///
+                    /// ```
+                    #[doc = "use scrypto::prelude::*;"]
+                    ///
+                    #[doc = "let n: I16  = I16::tfrom(0b0000000_01010101i16);"]
+                    #[doc = "assert_eq!(n, I16::tfrom(85i16));"]
+                    ///
+                    /// let m = n.reverse_bits();
+                    ///
+                    /// assert_eq!(m.0 as u16, 0b10101010_00000000);
+                    #[doc = "assert_eq!(m, I16::tfrom(-22016i16));"]
+                    /// ```
+                    #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+                    #[inline]
+                    pub fn reverse_bits(self) -> Self {
+                        $t(self.0.into_iter().rev().map(|x| x.reverse_bits()).collect::<Vec<u8>>().try_into().unwrap())
+                    }
+
+                    /// Converts an integer from big endian to the target's endianness.
+                    ///
+                    /// On big endian this is a no-op. On little endian the bytes are
+                    /// swapped.
+                    ///
+                    /// # Examples
+                    ///
+                    /// Please note that this example is shared between large integer types.
+                    /// Which explains why `I8` is used here.
+                    ///
+                    /// Basic usage:
+                    ///
+                    /// ```
+                    #[doc = "use scrypto::prelude::*;"]
+                    ///
+                    #[doc = "let n = " $t "::tfrom(0x1Ai8);"]
+                    ///
+                    /// if cfg!(target_endian = "big") {
+                    #[doc = "    assert_eq!(" $t "::from_be(n), n)"]
+                    /// } else {
+                    #[doc = "    assert_eq!(" $t "::from_be(n), n.swap_bytes())"]
+                    /// }
+                    /// ```
+                    #[inline]
+                    #[must_use]
+                    pub fn from_be(x: Self) -> Self {
+                        if cfg!(target_endian = "big") {
```

Comment:
Not sure if we should include this platform-dependent functions? Any use case?

Reply:
Just wanted to add for completeness. If you don't like it I go ahead and remove the whole thing.

Reply:
Removed in https://github.com/radixdlt/radixdlt-scrypto/pull/388/commits/5cbbaa3004ead5ee1a71b03a2c3ff6525526325f